shopfloor/services/location_content_transfer.py

```diff
@@ -0,0 +1,1194 @@
+                return self._response_for_start(
+                    message=self.msg_store.cannot_move_something_in_picking_type()
+                )
+
+        if not move_lines:
+            if not self.is_allow_move_create():
+                savepoint.rollback()
+                return self._response_for_start(
+                    message=self.msg_store.location_empty(location)
+                )
+            new_moves = self._create_moves_from_location(location)
+            if not new_moves:
+                savepoint.rollback()
+                return self._response_for_start(
+                    message=self.msg_store.location_empty(location)
+                )
+            new_moves._action_confirm(merge=False)
+            new_moves._action_assign()
+            if not all([x.state == "assigned" for x in new_moves]):
+                savepoint.rollback()
+                return self._response_for_start(
+                    message=self.msg_store.new_move_lines_not_assigned()
+                )
+            move_lines = new_moves.move_line_ids
+            for line in move_lines:
+                if not self.is_dest_location_valid(line.move_id, line.location_dest_id):
+                    savepoint.rollback()
+                    return self._response_for_start(
+                        message=self.msg_store.location_content_unable_to_transfer(
+                            location
+                        )
+                    )
+
+        stock = self._actions_for("stock")
+        if self.work.menu.ignore_no_putaway_available and stock.no_putaway_available(
+            self.picking_types, move_lines
+        ):
+            # the putaway created a move line but no putaway was possible, so revert
+            # to the initial state
+            savepoint.rollback()
+            return self._response_for_start(
+                message=self.msg_store.no_putaway_destination_available()
+            )
+
+        stock.mark_move_line_as_picked(move_lines)
+
+        unreserved_moves._action_assign()
+
+        savepoint.release()
+
+        return self._router_single_or_all_destination(move_lines.picking_id)
+
+    def _find_transfer_move_lines_domain(self, location):
+        return [
+            ("location_id", "=", location.id),
+            ("state", "in", ("assigned", "partially_available")),
+            ("qty_done", ">", 0),
+            # TODO check generated SQL
+            ("picking_id.user_id", "=", self.env.uid),
+        ]
+
+    def _find_transfer_move_lines(self, location):
+        """Find move lines currently being moved by the user"""
+        lines = self.env["stock.move.line"].search(
+            self._find_transfer_move_lines_domain(location)
+        )
+        return lines
+
+    # hook used in module shopfloor_checkout_sync
+    def _write_destination_on_lines(self, lines, location):
+        lines.location_dest_id = location
+        lines.package_level_id.picking_id.location_dest_id = location
```

Comment:
This line needs fixing. Each package taken from the source location can have their own destination